cmake/functions/four_c_auto_define_module.cmake

```diff
@@ -75,23 +75,46 @@ function(four_c_auto_define_module)
     endif()
   endforeach()
 
-  # Add the headers as a file set to the interface library. This will automatically add the current directory as a search directory.
-  # Since we append sources to a target successively, we need to add multiple file sets with a unique name. Use
-  # the current directory name for this.
-  # Replace non-alphanumeric characters with underscores and make the name lowercase. Drop any leading digits and underscores.
-  string(REGEX REPLACE "[^a-zA-Z0-9]" "_" _file_set_name "${CMAKE_CURRENT_SOURCE_DIR}")
-  string(TOLOWER "${_file_set_name}" _file_set_name)
-  string(REGEX REPLACE "^[0-9_]+" "" _file_set_name "${_file_set_name}")
-
+  # We want to export/install the base directory with hierarchy. At the same time we also want to include them
+  # without hierarchies at the target as 4C does.
   target_sources(
     ${_target}_deps
     INTERFACE FILE_SET
-              ${_file_set_name}
-              TYPE
               HEADERS
               FILES
               ${_headers}
+              BASE_DIRS
+              ${PROJECT_SOURCE_DIR}/src
     )
+  include(GNUInstallDirs)
+  file(RELATIVE_PATH _relative_path ${CMAKE_SOURCE_DIR}/src ${CMAKE_CURRENT_SOURCE_DIR})
+  if(NOT "${_relative_path}" STREQUAL "config")
```

Comment:
Please move this hack out of the otherwise general function. If we need to do something special for `config`, we can do that in its `CMakeLists.txt`.